staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -382,25 +382,22 @@ func (o *DebugOptions) visitPod(ctx context.Context, pod *corev1.Pod) (*corev1.P
 // debugByEphemeralContainer runs an EphemeralContainer in the target Pod for use as a debug container
 func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, string, error) {
 	pods := o.podClient.Pods(pod.Namespace)
-	ec, err := pods.GetEphemeralContainers(ctx, pod.Name, metav1.GetOptions{})
+	klog.V(2).Infof("existing ephemeral containers: %v", pod.Spec.EphemeralContainers)
+
+	debugContainer := o.generateDebugContainer(pod)
+	klog.V(2).Infof("new ephemeral container: %#v", debugContainer)
+
+	pod.Spec.EphemeralContainers = append(pod.Spec.EphemeralContainers, *debugContainer)
+	result, err := pods.UpdateEphemeralContainers(ctx, pod.Name, pod, metav1.UpdateOptions{})
 	if err != nil {
 		// The pod has already been fetched at this point, so a NotFound error indicates the ephemeralcontainers subresource wasn't found.
```

Comment:
> a NotFound error indicates the ephemeralcontainers subresource wasn't found.

or it means the pod was deleted in the meantime

Reply:
Hmmm, good point. It looks like StatusDetail will be set differently in this case. This seems a reasonable test given:

1. It will be removed in the future when we remove the feature gate.
2. I've added an integration test so we'll notice if it changes.
3. A false negative for detecting the feature being disabled will still result in a reasonably descriptive error message ("server could not find the specified resource")